### PR TITLE
Check for peers before broadcasting smart contract transactions

### PR DESCRIPTION
**Pull request by rowandh**
See 3893.

### CODE REVIEW

**Comment by zeptin** [comment]
@rowandh The tests seem to be failing

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs [hunk 1]
```diff
@@ -335,6 +339,12 @@ public IActionResult BuildAndSendCreateSmartContractTransaction([FromBody] Build
 
             Transaction transaction = this.network.CreateTransaction(response.Hex);
 
+            if (!this.connectionManager.ConnectedPeers.Any())
+            {
+                this.logger.LogTrace("(-)[NO_CONNECTED_PEERS]");
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.Forbidden, "Can't send transaction: sending transaction requires at least one connection!", string.Empty);
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: Not sure if the `HttpStatusCode` should be forbidden? I mean, we did infact make a valid API call, its just that an internal component wasn't able to provide the requested data? Http status codes is relevant to the call and not the contents of the it.

`HTTP Status Code 403: The server understood the request but refuses to authorize it.`

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: I would return an OK result with the message in it as the call "succeeded".

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs [hunk 2]
```diff
@@ -372,6 +382,12 @@ public IActionResult BuildAndSendCallSmartContractTransaction([FromBody] BuildCa
...
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: Sam here.

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs (hunk 1 above)
Comment (rowandh) [reply src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: I just borrowed the implementation from `WalletController`

Comment (zeptin) [reply src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: I don't disagree, but the normal wallet controller currently returns forbidden.

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/SmartContractsController.cs]: OK lets fix this in another PR.

**Pull request merged**